cukinia

```diff
@@ -621,6 +622,37 @@ _cukinia_mount()
 	return 0
 }
 
+# _cukinia_i2c: Check if i2c device exists
+# Optional: Check of a device address
+# arg1: the bus number
+# arg2: (optional) the device address
+_cukinia_i2c()
+{
+	local bus_number="$1"
+	local device_address="$2"
+	local ret
+
+	_cukinia_prepare "Checking if i2c-$bus_number${device_address:+-$device_address} exists"
+
+	if [ ! -d /sys/bus/i2c/devices/i2c-"$bus_number" ]; then
+		return 1
+	fi
+
+	if [ -z "$device_address" ]; then
+		return 0
+	fi
+
+	if [ ! "$(which i2cdetect)" ]; then
```

Comment:
`test ! "string"` returns true if "string" is false (empty), so you should either use -z for consistency with above, or maybe what you meant  
```
if [ ! -x "$(which i2cdetect)" ]; then   # note: watch out stderr

if [ -z "$(which i2cdetect)" ]; then # likewise